### PR TITLE
[#1252] custom Iceberg REST service default http and https port

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
If user doesn't set Iceberg REST service default http and https port, use a custom default one different from the default value for Gravitino server

### Why are the changes needed?

If not set the port of Iceberg, the server can't start
Fix: #1252 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
gravitino server could start with not setting  Iceberg REST server HTTP port

### CODE REVIEW

File: server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java [hunk 1]
```diff
@@ -25,6 +26,10 @@
 public final class JettyServerConfig {
   private static final Logger LOG = LoggerFactory.getLogger(JettyServerConfig.class);
   private static final String SPLITTER = ",";
+  public static final int DEFAULT_ICEBERG_REST_SERVICE_HTTP_PORT = 9001;
+  public static final int DEFAULT_ICEBERG_REST_SERVICE_HTTPS_PORT = 8434;
```
Comment (FANNG1) [inline server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java:30]: will be 9433

File: docs/iceberg-rest-service.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Deploy the Gravitino server to the `GRAVITINO_HOME` directory. You can find the
...
-| `gravitino.auxService.iceberg-rest.httpPort`                | The port of the Gravitino Iceberg REST catalog service.                                                                                                                                                                                                    | `8090`                                                                       | Yes      | 0.2.0         |
+| `gravitino.auxService.iceberg-rest.httpPort`                | The port of the Gravitino Iceberg REST catalog service.                                                                                                                                                                                                    | `9001`                                                                       | Yes      | 0.2.0         |
```
Comment (jerryshao) [inline docs/iceberg-rest-service.md:37]: Can you please also update security doc https related part.

File: server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java [hunk 3]
```diff
@@ -566,4 +574,12 @@ Set<String> getSupportedCipherSuites() {
     }
     return Sets.newHashSet(context.getServerSocketFactory().getSupportedCipherSuites());
   }
+
+  private static Map<String, String> overwriteJettyDefaultConfig(
+      Map<String, String> properties, OverwriteDefaultConfig config) {
+    if (config.getOverwriteDefaultConfig().isEmpty()) return properties;
+    Map<String, String> newProperties = new HashMap<>(properties);
+    config.getOverwriteDefaultConfig().forEach((k, v) -> newProperties.putIfAbsent(k, v));
```
Comment (jerryshao) [inline server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java:582]: Why use this `putIfAbsent`, I would assume that we should use put to overwrite?

Comment (FANNG1) [reply server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java:582]: If user set IcebergRESTservice http port explictly, we should not overwrite it

Comment (jerryshao) [reply server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java:582]: I see.

File: docs/iceberg-rest-service.md (hunk 2 above)
Comment (FANNG1) [reply docs/iceberg-rest-service.md:37]: done

File: server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java (hunk 1 above)
Comment (FANNG1) [reply server-common/src/main/java/com/datastrato/gravitino/server/web/JettyServerConfig.java]: done

**Review by jerryshao — APPROVED** [review]

**Pull request merged**